tests/test_python.py

```diff
@@ -207,113 +216,42 @@ def test_pip_install_source_cryptography(auto_container_per_test):
+    resp = requests.get(f"http://0.0.0.0:{host_port}")
+    resp.raise_for_status()
+    assert resp.text
 
-    # port status inspection with timeout
-    for _ in range(retry):
-        time.sleep(1)
-        portstatus = container_per_test.connection.socket(
-            f"tcp://0.0.0.0:{port}"
-        ).is_listening
 
-        if portstatus:
-            break
-
-    t3 = time.time() - t0
-
-    # check inspection success or timeout
-    assert portstatus, (
-        "Timeout expired: expected port not listening. Time marks: before "
-        + f"server start {t1}s, after start {t2}s, after {t} loops {t3}s."
-    )
-
-
-@pytest.mark.skipif(
-    OS_VERSION == "tumbleweed",
-    reason="pip --user not working due to PEP 668",
-)
-@pytest.mark.parametrize(
-    "container_per_test", CONTAINER_IMAGES_T2, indirect=["container_per_test"]
-)
 @pytest.mark.parametrize(
-    "destdir, appl2, url, xfilename",
-    [
-        (
-            BCDIR + OUTDIR,
-            "communication_examples.py",
-            "https://opensource.suse.com/bci/SLE_BCI_logomark_green.svg",
-            "SLE_BCI_logomark_green.svg",
-        )
-    ],
+    "container_per_test", REQUESTS_CONTAINER_IMAGES, indirect=True
 )
-def test_python_webserver_2(
-    container_per_test: ContainerData,
-    host,
-    container_runtime: OciRuntimeBase,
-    destdir: str,
-    appl2: str,
-    url: str,
-    xfilename: str,
-) -> None:
-    """Test that the python `wget <https://pypi.org/project/wget/>`_ library,
-    coded in the appl2 module, is able to fetch files from a webserver
-    """
-
-    # install wget for python
-    container_per_test.connection.run_expect([0], "pip install wget")
+def test_python_fetch_remote_file(container_per_test: ContainerData) -> None:
+    """Test that python's requests is able to fetch the SLE_BCI logo from
+    opensource.suse.com
 
-    # copy an application file from the local test-server into the running
-    # container under test
-    host.check_output(
-        f"{container_runtime.runner_binary} cp {ORIG + APPDIR + appl2} "
-        f"{container_per_test.container_id}:{BCDIR + APPDIR}",
-    )
-
-    # check the test python module is present in the container
-    assert container_per_test.connection.file(BCDIR + APPDIR + appl2).is_file
-
-    # check expected file not present yet in the destination
-    assert not container_per_test.connection.file(destdir + xfilename).exists
+    """
+    _LOGO_URL = "https://opensource.suse.com/bci/SLE_BCI_logomark_green.svg"
+    logo_resp = requests.get(_LOGO_URL)
+    logo_resp.raise_for_status()
 
-    # execution of the python module in the container
-    bci_python_wget = container_per_test.connection.check_output(
-        f"timeout --preserve-status 120s python3 {APPDIR + appl2} {url} {destdir}",
+    assert (
+        container_per_test.connection.check_output(
+            f"source .venv/bin/activate; python -c \"import requests; print(requests.get('{_LOGO_URL}').text)\""
```

Comment:
The timeout=120 from previous code is lost now, isn't it?

Reply:
I've added it back